azure/converters/managedagentpool.go

```diff
@@ -25,37 +31,50 @@ import (
 func AgentPoolToManagedClusterAgentPoolProfile(pool *asocontainerservicev1hub.ManagedClustersAgentPool) asocontainerservicev1hub.ManagedClusterAgentPoolProfile {
 	properties := pool.Spec
 	agentPool := asocontainerservicev1hub.ManagedClusterAgentPoolProfile{
-		Name:                        ptr.To(pool.AzureName()),
-		VmSize:                      properties.VmSize,
-		OsType:                      properties.OsType,
-		OsDiskSizeGB:                properties.OsDiskSizeGB,
-		Count:                       properties.Count,
-		Type:                        properties.Type,
-		OrchestratorVersion:         properties.OrchestratorVersion,
-		VnetSubnetReference:         properties.VnetSubnetReference,
-		Mode:                        properties.Mode,
-		EnableAutoScaling:           properties.EnableAutoScaling,
-		MaxCount:                    properties.MaxCount,
-		MinCount:                    properties.MinCount,
-		NodeTaints:                  properties.NodeTaints,
-		AvailabilityZones:           properties.AvailabilityZones,
-		MaxPods:                     properties.MaxPods,
-		OsDiskType:                  properties.OsDiskType,
-		NodeLabels:                  properties.NodeLabels,
-		EnableUltraSSD:              properties.EnableUltraSSD,
-		EnableNodePublicIP:          properties.EnableNodePublicIP,
-		NodePublicIPPrefixReference: properties.NodePublicIPPrefixReference,
-		ScaleSetPriority:            properties.ScaleSetPriority,
-		ScaleDownMode:               properties.ScaleDownMode,
-		SpotMaxPrice:                properties.SpotMaxPrice,
-		Tags:                        properties.Tags,
-		KubeletDiskType:             properties.KubeletDiskType,
-		LinuxOSConfig:               properties.LinuxOSConfig,
-		EnableFIPS:                  properties.EnableFIPS,
-		EnableEncryptionAtHost:      properties.EnableEncryptionAtHost,
-	}
-	if properties.KubeletConfig != nil {
-		agentPool.KubeletConfig = properties.KubeletConfig
+		AvailabilityZones:                 properties.AvailabilityZones,
```

Comment:
Most of the new fields here aren't possible to set from an AzureManagedMachinePool, so there's a fair bit of no-op here. I was too lazy to audit which ones can actually be set, but keeping everything here makes for one less thing to worry about in case we do add or remove any features from CAPZ.

Reply:
ACK, thx for adding the maintenance comments so we're less likely to omit this in the future